Update dbt tutorials to use DbtProject (#23042)
dagster-io/dagster: docs/content/integrations/dbt/using-dbt-with-dagster/load-dbt-models.mdx

In our `definitions.py` Python file, we import from `assets.py`, which contains the code to model our dbt models as Dagster assets. To return a Dagster asset for each dbt model, the code in this `assets.py` file needs to know what dbt models you have. It finds out what models you have by reading a file called a `manifest.json`, which is a file that dbt can generate for any dbt project and contains information about every model, seed, snapshot, test, etc. in the project.

To retrieve the `manifest.json`, `assets.py` imports from `project.py`, which defines an internal representation of your dbt project. Then, in `assets.py`, the path to the `manifest.json` file can be accessed with `dbt_project.manifest_path`:

```python file=integrations/dbt/tutorial/load_dbt_models/project.py startafter=start_load_project endbefore=end_load_project
from pathlib import Path

from dagster_dbt import DbtProject

dbt_project = DbtProject(
    project_dir=Path(__file__).joinpath("..", "..", "..").resolve(),
)

> # If `dagster dev` is used, the dbt project will be prepared at run time, and manifest will be created.

Reviewer: ```suggestion
# If `dagster dev` is used, the dbt project will be prepared to create the manifest at run time.
```

Author: All changes made in [633ba87](https://github.com/dagster-io/dagster/pull/23042/commits/633ba87b5024b8a12fda6df9cc1ff20a395183fc)

# Otherwise, we expect a manifest to be present in the project's target directory.
dbt_project.prepare_if_dev()
```

> As the comment explains, calling `dbt_project.prepare_if_dev()` prepares your dbt project during development, i.e. when `dagster dev` is used, which ensures that the dbt project is ready to be used by Dagster. This process creates the `manifest.json` file.

Reviewer: ```suggestion
Invoking `dbt_project.prepare_if_dev()` prepares your dbt project for local development by creating the `manifest.json` at run time in `dagster dev`.

> To disable the project preparation during development, `dbt_project.prepare_if_dev()` can be removed.

Reviewer: ```suggestion
To disable this behavior during local development, remove `dbt_project.prepare_if_dev()`.
```
